### PR TITLE
feat: playlist management

### PR DESCRIPTION
**Pull request by SinchanaMS**
This request adds the following features:

- The user can create playlists
- Add videos to multiple playlists
- Delete the playlist

### Preview:

1. Create playlist:
image

2. Empty playlist:
image

3. Playlist with videos:
image

4. Playlist page:
image

### CODE REVIEW

File: src/router/utils/HelperFunctions.jsx [hunk 1]
```diff
@@ -186,6 +186,127 @@ export const clearHistory = async (videoDispatch) => {
...
+      { video },
+      {
+        headers: {
+          authorization: userToken,
+        },
+      }
+    );
+    if (response.status === 409) {
+      Toast({ type: "info", message: "Video already exists in the playlist" });
+    }
```
Comment (srejitk) [inline src/router/utils/HelperFunctions.jsx:263]: Good work providing a toast for this

**Review by srejitk — APPROVED** [review]
Good work! The styling is really minimal and clean.
Really liked how you've segregated and modularised everything.

File: src/reducers/videoReducer.jsx [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+      return { ...state, history: action.payload };
     case "CLEAR_HISTORY":
-      return { ...videoData, history: action.payload };
+      return { ...state, history: action.payload };
+    case "CREATE_PLAYLIST":
+      return { ...state, playlists: action.payload };
+    case "DELETE_PLAYLIST":
+      return { ...state, playlists: action.payload };
+    case "ADD_TO_PLAYLIST": {
+      let updatedPlaylist = state.playlists.map((item) =>
```
Comment (Debarshi95) [inline src/reducers/videoReducer.jsx]: Prefer using const if variable is not updated

File: src/reducers/videoReducer.jsx [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+        item._id === action.payload._id ? action.payload : item
+      );
+      return { ...state, playlists: updatedPlaylist };
+    }
+    case "DELETE_FROM_PLAYLIST":
+      return state;
     default:
-      return videoData;
+      return state;
```
Comment (Debarshi95) [inline src/reducers/videoReducer.jsx:32]: default and DELETE_FROM_PLAYLIST both return the state??

**Review by Debarshi95 — APPROVED** [review]
Couldn't find any major issue. 
Liked the way you destructured items from objects. Add some minor comments, may be fix them in next PR.
Good Work. You can continue adding remaining features

File: src/router/utils/HelperFunctions.jsx (hunk 1 above)
Comment (SinchanaMS) [reply src/router/utils/HelperFunctions.jsx:263]: Thank you!

File: src/reducers/videoReducer.jsx (hunk 2 above)
Comment (SinchanaMS) [reply src/reducers/videoReducer.jsx]: Thanks, I've changed it.

File: src/reducers/videoReducer.jsx (hunk 3 above)
Comment (SinchanaMS) [reply src/reducers/videoReducer.jsx:32]: I missed changing this, thank you for pointing it out!

**Comment by SinchanaMS** [comment]
Thank you for the reviews. Made the necessary changes, merging to dev.

**Pull request merged**